[U-3728] Add `expiration_policy_id` attribute to `betteruptime_monitor`

dmajda opened the pull request:
The attribute reflects the same attribute in the Monitor API, which was recently added.

Per discussion with @PetrHeinz, `expiration_policy_id` is not computed so it is possible to set it to `null` by users. We also always send its value to the API on create/update, even when it is not set or changed. This is done because we use Go structs + default JSON serialization, meaning it's impossible to differentitate between not setting the attribute and setting it to `null`.

PetrHeinz reviewed (approved):
Thanks 🙌 

Could you please add it to `examples/advanced` and into the `resource_monitor_test`? 🙏  You might want test specifically the null <-> 0 conversion. The `TestCheckCalledRequest` method might be useful for asserting API requests.

dmajda committed:
Use `TestServer` in `resource_monitor_test.go`

This will allow us to use `TestCheckCalledRequest` in tests.

dmajda committed:
Add `expiration_policy_id` attribute to `betteruptime_monitor`

The attribute reflects the same attribute in the Monitor API, which was
recently added.

Per discussion with @PetrHeinz, `expiration_policy_id` is not computed
so it is possible to set it to `null` by users. We also always send its
value to the API on create/update, even when it is not set or changed.
This is done because we use Go structs + default JSON serialization,
meaning it's impossible to differentitate between not setting the
attribute and setting it to `null`.

dmajda committed:
Add `expiration_policy_id` to the advanced example

PetrHeinz reviewed (approved):
Thanks 🙌  Looks great!

💄 When adding a newly added thing into examples, I like to bump the version in relevant `versions.tf` and the `Makefile`, as to make sure anyone running the example will have provider that actually supports the feature.

💄 Please don't force push if you can avoid it - it easier to do a re-review on just the new changes. Also, Github actions history is kept.

💡 To release a new Terraform provider, you just need to merge and create a new release in Github. Autogenerated release notes are fine. It will push a tag, and Github Actions should take care of the rest.

dmajda committed:
Bump version to 0.18.1
